Windows 64.

kntroh opened the pull request:
Fixes for Windows 64-bit.

I tested all snippets and my application on Windows 7 64-bit with VS 2013 and Windows SDK for 8.1.

jacob-carlborg commented on README.markdown:
````diff
@@ -87,6 +91,14 @@ For Windows:
+  If you use Microsoft linker:
+
+  ```
+  $ dmd main.d -I<dwt>\imp -J<dwt>\res -L/LIBPATH:<dwt>\lib ^
````
The -J flag doesn't match the one for DMC linker.

kntroh replied:
`-J<dwt>\res` is required for compiler. dmd reads `res\org.eclipse.swt.internal.SWTMessages_*.properties` when compile time.

jacob-carlborg replied:
Yes, but the existing instructions says `-J<dwt>\org.eclipse.swt.win32.win32.x86\res`.

kntroh replied:
The contents of the folder is copied to `<dwt>\res` when building.

Examples of 32-bit and 64-bit seems to right both, What had better fix either?

jacob-carlborg replied:
I don't know which is best, but I should be consistent.

kntroh replied:
I think `-J<dwt>\res` is proper maybe.

I'll modify an example for optlink.

kntroh replied:
I updated it.

jacob-carlborg commented on build.d:
```diff
@@ -321,13 +324,26 @@ void buildApp( string basedir, string srcdir, string resdir, in string[] dflags,
-        rsp ~= win_path(.absolutePath("win-res\\resource.res"));
-        foreach (libname; libnames) {
-            rsp ~= "-L+" ~ libname ~ LIBEXT;
+        if (extraOptions.countUntil("-m64") == -1) {
```
I think `canFind` is more appropriate.

kntroh replied:
I'll replace it.

jacob-carlborg commented:
Would it be difficult to extend this build script to support 32bit COFF?

kntroh commented:
> Would it be difficult to extend this build script to support 32bit COFF? 

I think that it is possible, but need a new option. (Windows dmd user uses OPTLINK normally.)

kntroh commented:
I updated pull request.

jacob-carlborg commented:
> I think that it is possible, but need a new option. (Windows dmd user uses OPTLINK normally.)

I think we can use the same flag as DMD uses, `-m32mscoff`.

etcimon commented:
I actually have a dub.json file on my fork that supports 32 bit ms coff and integrates nicely to dub projects. It needs removal of the pragma instructions in the code and doesn't have resource compiler configured yet but I really needed it for my projects if interested: https://github.com/etcimon/dwt/blob/master/dub.json

etcimon commented:
How do you avoid the errors related to int-long implicit conversion? I tried to compile with -m64 but couldn't due to these

kntroh commented:
I am trying to support for ms32coff now. but more difficult than I thought, take time.

> How do you avoid the errors related to int-long implicit conversion? I tried to compile with -m64 but couldn't due to these

It is brute force. I fixed all a signature of API functions, and removed compile errors. then checked all cast, and tested all snippets and DWT applications, and I fixed occured problem.

etcimon commented:
> It is brute force. I fixed all a signature of API functions, and removed compile errors. then checked all cast, and tested all snippets and DWT applications, and I fixed occured problem.

I did that too it took hours but then winapi failed at createHandle. Can you share a link to this?

etcimon commented:
Ok i found the other pull. Thanks for the nice work

etcimon commented:
> I am trying to support for ms32coff now. but more difficult than I thought, take time.

You need to edit win64.mak. I changed it to MODEL=32mscoff and remove all occurence of amd64/ in the file (there are 3), for both druntime and phobos. Save this to win32mscoff.mak 

You need to place the phobos32mscoff.lib into dmd2/windows/lib32mscoff/ (the folder doesn't exist).

kntroh commented:
I updated this pull request for support for 32bit COFF now.

Thanks @etcimon.

jacob-carlborg commented:
@kntroh could you please mention the `-m32mscoff` flag in the documentation.

kntroh commented:
Added it.

jacob-carlborg commented:
@kntroh Awesome work as always, thanks.
